DirectEmployers/MyJobs
# Ensure missing forward slashes don't break stuff [PD-3005]

## What's this PR do?
Redirect has the concept of a canonical microsite (and the CanonicalMicrosite model). In the past, a requirement of use was that the canonical microsite url end in a forward slash. More recently, people stopped doing that. This PR makes it so that a missing slash doesn't break.
## Screenshots (if appropriate)
N/A
## Where should the reviewer start?
The diff is fairly short - I think you can find it.
## How should this be manually tested?
10. Go to your admin and make some canonical microsites with urls ending in slashes or not.
20. Make some Redirect entries (must be through a Django shell).
30. Navigate to my.jobs/<guid>+ and ensure that the line denoted by "RetLink=" appears to be something that would resolve to the canonical microsite that you created.
40. GOTO 10, choosing the option not chosen first time.
## Any background context you want to provide?
```
In [1]: from urlparse import urljoin

In [2]: urljoin('https://foo.jobs/', 'whatever')
Out[2]: 'https://foo.jobs/whatever'

In [3]: urljoin('https://foo.jobs/', '/whatever')
Out[3]: 'https://foo.jobs/whatever'

In [4]: urljoin('https://foo.jobs', 'whatever')
Out[4]: 'https://foo.jobs/whatever'

In [5]: urljoin('https://foo.jobs', '/whatever')
Out[5]: 'https://foo.jobs/whatever'
```
